lib/cloudregister/registerutils.py

```diff
@@ -924,6 +937,19 @@ def has_region_changed(cfg):
     return True
 
 
+# ----------------------------------------------------------------------------
+def has_rmt_in_hosts(server):
+    """Check if an entry for the given update server is in the hosts file"""
+    hosts_content = open('/etc/hosts').read()
+    srv_ipv4 = server.get_ipv4()
+    srv_ipv6 = server.get_ipv6()
+
+    if srv_ipv4 in hosts_content or srv_ipv6 in hosts_content:
+        return True
+
+    return False
+
+
```

Comment:
Unlike as ipv4, ipv6 comparisons can be a bit tricky because of the different ways ipv6 can be written.

For example all these IPs are the same but written differently:
```
11:22:33:44:0:0:0:55
0011:0022:0033:0044:0000:0000:0000:0000:0055
11:22:33:44::55
```
If we want to match IPv6, parsing the string into a ipaddress struct is better approach.